lms/views/lti/basic_launch.py

```diff
@@ -291,15 +275,20 @@ def _do_launch(self, document_url, grading_supported=True):
         if grading_supported:
             self.context.js_config.maybe_enable_grading()
 
+        self.context.js_config.enable_lti_launch_mode()
+        self.context.js_config.maybe_set_focused_user()
```

Comment:
`maybe_set_focused_user` appears to be grading related, but it's actually Speedgrader related. Which in our world is `grading_supported=False`. This is only for Canvas.

Reply:
This is a bit of a mess but I wouldn't get into refactoring it now.

We very explicitly say here `grading_supported` but the we call all of this `maybe_` methods that do check themselves if that's the case.

`js_config.add_document_url` calls, unconditionally, self._add_canvas_speedgrader_settings (which should be renamed to have the `maybe_?).

This means that for canvas_files, the only place we pass `grading_supported=False` we don't call `maybe_enable_grading()` (that checks itself if we are in canvas) but for the rest, for all non-canvas launches we call `add_canvas_speedgrader_settings` that also check if we are in canvas.

I think once we try to tackle recording if assignment are gradable etc, we should unify "maybe_enable_grading", call it all the time and have it deal with this differences.


For this PR I think what we are doing is fine.

Reply:
There's a separate PR to try and deal with the `maybe_` world: https://github.com/hypothesis/lms/pull/4050